support-frontend/assets/pages/contributions-landing/components/StripePaymentRequestButton/StripePaymentRequestButton.jsx

```diff
@@ -260,26 +243,29 @@ function onPayment(
   // the backend job finishes.
   paymentRequestComplete('success');
 
-  // We need to do this so that we can offer marketing permissions on the thank you page
+  // We always need an email address to do ecommerce on support.theguardian.com
   updatePayerEmail(paymentRequestData, props.updateEmail);
 
-  const isUsOrCanadian = props.countryGroupId === UnitedStates || props.countryGroupId === Canada;
-
-  const stateOrProvinceUpdateOk = isUsOrCanadian ?
-    updatePayerStateOrProvince(stateOrProvinceFromCard, props.stateOrProvince, props.updateStateOrProvince) : true;
-
-  // We need to update the country so it matches the state taken from the card
-  if (isUsOrCanadian && billingCountryFromCard) {
-    const isoCountry = countryFromString(billingCountryFromCard);
-    if (isoCountry) {
-      props.updateBillingCountry((isoCountry));
+  // Single doesn't need a name, but recurring (i.e. Zuora and Salesforce) needs a non-empty first and last name.
+  const nameValueOk: boolean = props.contributionType === 'ONE_OFF' ||
+    updatePayerName(paymentRequestData, props.updateFirstName, props.updateLastName);
+
+  // Single doesn't need a state, however recurring (i.e. Zuora) needs a valid state for US and CA billing countries.
+  const validatedCountryFromCard: Option<IsoCountry> = findIsoCountry(billingCountryFromCard);
+  let countryAndStateValueOk = props.contributionType === 'ONE_OFF' || !['US', 'CA'].includes(validatedCountryFromCard);
+  if (validatedCountryFromCard) {
+    props.updateBillingCountry(validatedCountryFromCard);
+    const validatedBillingStateOrProvinceFromCard: Option<StateProvince> =
+      stateProvinceFromString(validatedCountryFromCard, billingStateOrProvinceFromCard);
+    if (validatedBillingStateOrProvinceFromCard) {
+      props.updateStateOrProvince(validatedBillingStateOrProvinceFromCard);
+      countryAndStateValueOk = true;
+    } else if (!props.stateOrProvince) {
+      logException('Missing address_state in payment request token and no state/province selected in the form');
```

Comment:
this error will be logged even if a state/province is not required, e.g. a UK user

Reply:
Ah yes, before it only logged if US and Canada. I'll raise a fix.

Reply:
Fix is here: https://github.com/guardian/support-frontend/pull/2378